src/main.c

```diff
@@ -65,46 +66,40 @@ extern operation operations[];
 
 int main(int argc, char *argv[])
 {
-
     // Get command line options to hide parts of the display
-    if (argc >= 2 && argv[1][0] == '-') {
-
-        for (int i=1; argv[1][i] != '\0'; i++) {
-
-            switch (argv[1][i]) {
-
-                case 'h':
-                    history_enabled = 0;
-                    break;
-
-                case 'b':
-                    binary_enabled = 0;
-                    break;
-
-                case 'x':
-                    hex_enabled = 0;
-                    break;
-
-                case 'd':
-                    decimal_enabled = 0;
-                    break;
-
-                case 'o':
-                    operation_enabled = 0;
-                    break;
-
-                case 's':
-                    symbols_enabled = 0;
-                    break;
-		default:
-		    fprintf(stderr, "Unrecognized option: -%c\n", argv[1][i]);
-		    exit(EXIT_FAILURE);
-
-            }
-
+    int opt;
+    while ((opt = getopt(argc, argv, "hbxdos")) != -1) {
```

Comment:
You should change `hbxdos` to `:hbxdos` because then we can check for unknown flags much better with the `?` case.
https://www.tutorialspoint.com/getopt-function-in-c-to-parse-command-line-arguments

Reply:
I could do this, however if you omit the ':' in the optstring, then the default behaviour for an unrecognised flag is to output the message `pcalc: invalid option -- 'option'` which emulates what this project currently does

Reply:
This sounds good, @DevManu-de can I resolve the conflicts and merge?

Reply:
If I add the `:` it just exits without resonding `pcalc: invalid option -- 'option'`

Reply:
Ok mt fault please excuse.

Reply:
No problem. I've merged the pull request and tested it successfully. Thank you both :)